### DISCUSSION TITLE
Can Synthetic vertices be removed or disabled？

### DISCUSSION THREAD

**Discussion by sonw7** [gwlucastrig/Tinfour#100] [Q&A]
These points themselves have no issues, but for personal reasons, they have a significant impact on the next step of processing the generated data.So I wonder if this feature can be avoided while adding constraints？

**Comment by gwlucastrig** [comment]
First off, I apologize for the delay in answering you.  I did not see your posting until this morning.

Currently, the synthetic vertices are only introduced when you add constraints.  The vertices are needed to ensure that the triangulation preserves the Delaunay criterion.  However, you can turn off this feature if you want.   In the Incremental Tin classes, the call to "addConstraints" takes two arguments.  The second, "restoreConformity" controls whether synthetic points are generated.  For more information, see https://gwlucastrig.github.io/TinfourDocs/javadoc/TinfourCore-2.1.6-javadoc/index.html

To illustrate why we might want to add the synthetic points, I attached a picture from the web article [What is the Constrained Delaunay](https://gwlucastrig.github.io/TinfourDocs/DelaunayIntroCDT/index.html).   When you add a constraint, it can introduce triangles that do not conform to the Delaunay criterion.  But we don't have to add the synthetic vertices unless we want to triangulation to remain Delaunay-compliant.

![CDT_Example_0](https://github.com/gwlucastrig/Tinfour/assets/19698145/73d8eee9-595d-48b1-8045-b46c8091a4fa)

**Reply by sonw7** [reply]
Thank you very much for your reply, it has been very helpful to me. I hope I can become a powerful person like you.

**Comment by gwlucastrig** [comment]
There is another special case where artificial points might occur.  If an application tries to insert two vertices that are too close together (or exactly coincident) into the triangulation, the Incremental TIN classes create a "merged vertex" to combine them into a single point.  By too close, it is usually one part in one million of the "significant point spacing" for the TIN.   When a pair or cluster of points get that close together, numerical issues start to degrade the algorithm.  So we treat them as a single point.

**Comment by gwlucastrig** [comment]
Thank you for your interest in the Tinfour software.

I wish you success in your endeavors.

Gary
